src/GNNGraphs/transform.jl

```diff
@@ -219,9 +219,16 @@ function remove_nodes(g::GNNGraph{<:COO_T}, nodes_to_remove::AbstractVector)
     w = get_edge_weight(g)
     edata = g.edata
     ndata = g.ndata
-
-    edges_to_remove_s = findall(x -> nodes_to_remove[searchsortedlast(nodes_to_remove, x)] == x, s)
-    edges_to_remove_t = findall(x -> nodes_to_remove[searchsortedlast(nodes_to_remove, x)] == x, t)
+    
+    function find_edges_to_remove(nodes, nodes_to_remove)
+        return findall(node_id -> begin
+            idx = searchsortedlast(nodes_to_remove, node_id)
+            idx >= 1 && idx <= length(nodes_to_remove) && nodes_to_remove[idx] == node_id
```

Comment:
```suggestion
            idx >= 1 && idx <= length(nodes_to_remove)
```

Reply:
This change results in tests failing

Reply:
I have re added it to see if tests pass

Reply:
why is that? if the idx returned from `searchsortedlast` is in the range then nodes_to_remove[idx] == node_id should be true

Reply:
`searchsortedlast` doesn't guarantee `nodes_to_remove[i] == node_id` since it only guarantees that node_id is not less than `nodes_to_remove[i]`. 

The condition is necessary to ensure that `node_id` is actually present in `nodes_to_remove`, rather than just being greater than or equal to the last element smaller than `node_id` in the sorted array.

I think this leads to false positives and thus the test failures.